docs/signing_helper.md

```diff
@@ -0,0 +1,303 @@
+    ```
+
+    Note down the digest algorithm “SHA256” and key name.
+
+    ```shell
+    export KMS_DIGEST_ALG=SHA256
+    export KMS_KEY_NAME=projects/$PROJECT_ID/locations/global/keyRings/my-key-ring-1/cryptoKeys/my-signing-key-1/cryptoKeyVersions/1
+    ```
+
+5. Pick a note name.
+
+    All attestations need to be attached to a note. The signer tool will automatically create a note for a given name. It can also reuse an existing note.
+
+    ```shell
+    export NOTE_ID=my-signer-note
+    export NOTE_NAME=projects/${PROJECT_ID}/notes/${NOTE_ID}
+    ```
+
+6. Create vulnerability signing policy.
+
+    An example policy is in the samples.
+
+    ```shell
+    cat samples/signer/policy.yaml
+
+    apiVersion: kritis.grafeas.io/v1beta1
+    kind: VulnzSigningPolicy
+    metadata:
+      name: my-vsp
+    spec:
+      imageVulnerabilityRequirements:
+        maximumFixableSeverity: MEDIUM
+        maximumUnfixableSeverity: MEDIUM
+        allowlistCVEs:
+        - projects/goog-vulnz/notes/CVE-2020-10543
+        - projects/goog-vulnz/notes/CVE-2020-10878
+        - projects/goog-vulnz/notes/CVE-2020-14155
+    ```
+
+7. Run signer on a built image (pass example).
+
+    1. Build and push an example good image.
+
+        ```shell
+        docker build -t gcr.io/$PROJECT_ID/signer-test:good -f samples/signer/Dockerfile.good .
+        docker push gcr.io/$PROJECT_ID/signer-test:good
+        ```
+
+    2. Note down the image digest url.
+
+        ```shell
+        export GOOD_IMG_URL=$(docker image inspect gcr.io/$PROJECT_ID/signer-test:good --format '{{index .RepoDigests 0}}')
+        ```
+
+    3. Run the signer.
+
+        ```shell
+        ./signer \
+          -v=10 \
+          -alsologtostderr \
+          -image=$GOOD_IMG_URL \
+          -policy=samples/signer/policy.yaml \
+          -kms_key_name=$KMS_KEY_NAME \
+          -kms_digest_alg=$KMS_DIGEST_ALG \
+          -note_name=$NOTE_NAME
+        ```
+
+        The signer should print out that the image "passes VulnzSigningPolicy my-vsp" and that an attestation is created and uploaded.
+
+8. Run signer on a built image (fail example).
+
+    1. Build and push an example good image.
+
+        ```shell
+        docker build -t gcr.io/$PROJECT_ID/signer-test:bad -f samples/signer/Dockerfile.bad .
+        docker push gcr.io/$PROJECT_ID/signer-test:bad
+        ```
+
+    2. Note down the image digest url.
+
+        ```shell
+        export BAD_IMG_URL=$(docker image inspect gcr.io/$PROJECT_ID/signer-test:bad --format '{{index .RepoDigests 0}}')
+        ```
+
+    3. Run the signer.
+
+        ```shell
+        ./signer \
+          -v=10 \
+          -alsologtostderr \
+          -image=$BAD_IMG_URL \
+          -policy=samples/signer/policy.yaml \
+          -kms_key_name=$KMS_KEY_NAME \
+          -kms_digest_alg=$KMS_DIGEST_ALG \
+          -note_name=$NOTE_NAME
+        ```
+
+        The signer should print out that the image "does not pass VulnzSigningPolicy my-vsp".
```

Comment:
Should we also provide examples of other modes?

Reply:
Ack.